### PR TITLE
Alerting: Integration test rule creation

### PR DESCRIPTION
**Pull request by gotjosh**
I discovered a few things while trying to integration test the rule creation flow - putting these up for debate so that we can tinker a bit on how this behaviour and payload look like.

Hopefully, this serves as a base to finish the rest of the endpoints and avoid regressions.

### CODE REVIEW

File: pkg/services/ngalert/api/api_ruler.go [hunk 1]
```diff
@@ -187,6 +191,11 @@ func (srv RulerSrv) RoutePostNameRulesConfig(c *models.ReqContext, ruleGroupConf
 	// TODO check quota
 	// TODO validate UID uniqueness in the payload
 
+	//TODO: Should this belong in alerting-api?
+	if ruleGroupConfig.Name == "" {
```
Comment (gotjosh) [inline pkg/services/ngalert/api/api_ruler.go:195]: This validates that rule groups must have a name. Is his correct?

File: pkg/services/ngalert/api/api_ruler.go [hunk 2]
```diff
@@ -43,8 +43,12 @@ func (srv RulerSrv) RouteDeleteRuleGroupConfig(c *models.ReqContext) response.Re
 	}
 	ruleGroup := c.Params(":Groupname")
 	if err := srv.store.DeleteRuleGroupAlertRules(c.SignedInUser.OrgId, namespace.Uid, ruleGroup); err != nil {
-		return response.Error(http.StatusInternalServerError, "failed to delete group alert rules", err)
```
Comment (gotjosh) [inline pkg/services/ngalert/api/api_ruler.go:46]: The validation below was a result of this - previously this endpoint would return everything OK deleted when we deleted nothing.

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 3]
```diff
@@ -50,3 +60,246 @@ func TestAlertAndGroupsQuery(t *testing.T) {
...
+		resp, err := http.Get(u)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+		b, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+
+		assert.Equal(t, resp.StatusCode, 202)
+		assert.JSONEq(t, `
+{
+   "default":[
```
Comment (gotjosh) [inline pkg/tests/api/alerting/api_alertmanager_test.go:154]: I'd like reviewers to pay special attention to this payload, does this seem correct? Do present or absent fields here look acceptable?

My general feeling is that fields like `expr`, `no_data_state` or `exec_err_state` should not empty but maybe this is already something that's in our roadmap?

Comment (papagian) [reply pkg/tests/api/alerting/api_alertmanager_test.go:161]: `expr` is not used for grafana managed alerts
for the other two we have to set to defaults values (`NoData` and `AlertingErrState` respectively) if they are missing.

File: pkg/services/ngalert/api/api_ruler.go (hunk 1 above)
Comment (gotjosh) [reply pkg/services/ngalert/api/api_ruler.go:195]: We have an endpoint that allows users to delete rules by rule group and namespace - without this becomes unusable.

File: pkg/tests/api/alerting/api_alertmanager_test.go (hunk 3 above)
Comment (gotjosh) [reply pkg/tests/api/alerting/api_alertmanager_test.go:161]: Should `expr` be `omitempty` then in `alerting-api`?

**Review by papagian — APPROVED** [review]
Great work with the tests

**Pull request merged**